tools/compress_assets.py

```
@@ -0,0 +1,141 @@
+
+
+def compress_file(file_path):
+    """Read file and compress with gzip (level 9)"""
+    with open(file_path, "rb") as f:
+        data = f.read()
+
+    # Compress with maximum compression (mtime=0 for deterministic output)
+    compressed = gzip.compress(data, compresslevel=9, mtime=0)
+
+    # Calculate hash for versioning/ETag
+    file_hash = hashlib.md5(compressed).hexdigest()[:8]
```

Ruff (0.15.2)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)